src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -56,11 +58,27 @@
       ;; Respond to all CSR validation failures with a 400
       (middleware-utils/plain-response 400 msg))))
 
+      ;;A recipient MUST ignore If-Modified-Since if the request contains an
+      ;; If-None-Match header field;
 (defn handle-get-certificate-revocation-list
-  [{:keys [cacrl]}]
-  (-> (ca/get-certificate-revocation-list cacrl)
-      (rr/response)
-      (rr/content-type "text/plain")))
+  "Always return the crl if no 'if-modified-since' header is provided or if that header is not in correct http-date format.
+  If the header is present and correctly formatted, only return the crl if the master crl is newer than the agent crl."
+  [request {:keys [cacrl]}]
+  (let [agent-crl-last-modified-val (rr/get-header request "If-Modified-Since")
+        agent-crl-last-modified-date-time (try
+                                            (time-format/parse
+                                              (time-format/formatters :rfc822)
+                                              (string/replace agent-crl-last-modified-val #"GMT" "+0000"))
+                                            (catch Exception e
```

Comment:
If there is no header sent, an exception will be raised with the string/replace method.  This seemed the driest implementation I could think of (considering I want to send the crl if there is no header, or if the header is malformed in any way from http-date format).  However, I know catching exceptions is kind of messy, especially since sending a nil value for this header is legit.  Let me know if you'd like me to check specifically for nil value instead of raising an error.

Reply:
I do find this a little hard to parse; its unclear without your note where the exception is coming from. I'd be in favor of making this more explicit somehow.

Reply:
I've reformatted it...not as dry, but easier to understand now I think.